server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -536,8 +570,21 @@ public ReplicationStats execute(Client client) throws Exception {
 
       // If we have some edits to send
       if (0 < edits.walEdits.getEditsSize()) {
+        // Check if we are interrupted before to writing the edits
+        if (Thread.interrupted()) {
+          log.debug("Replication work interrupted before writing edits, returning empty replication stats");
+          return new ReplicationStats(0L, 0L, 0L);
+        }
+
         log.debug("Sending {} edits", edits.walEdits.getEditsSize());
         long entriesReplicated = client.replicateLog(remoteTableId, edits.walEdits, tcreds);
```

Comment:
I believe so.  It is using `ThriftUtil#getClientNoTimeout` to get a client for sending the mutations to the peer server.